### PR TITLE
removed option from hash of eth block struct

### PR DESCRIPTION
**Pull request by dhaileytaha**
PR against origin/dev

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Your build is failing because you need to adapt all the existing call sites that at the moment dealt with the `Option` wrapper. In most places, that should simplify things because you can just use the existing field without calling `unwrap` or `expect` on it.

File: cnd/src/btsieve/ethereum.rs [hunk 1]
```diff
@@ -202,8 +201,7 @@ where
         match block_generator.async_resume().await {
             GeneratorState::Yielded(block) => {
                 let block_hash = block
-                    .hash
-                    .ok_or_else(|| anyhow::anyhow!("block without hash"))?;
+                    .hash;   // .ok_or_else(|| anyhow::anyhow!("block without hash"))?;
```
Comment (thomaseizinger) [inline cnd/src/btsieve/ethereum.rs:204]: Please remove the comment and inline the variable.

File: cnd/src/btsieve/ethereum.rs [hunk 2]
```diff
@@ -146,8 +145,8 @@ where
         match block_generator.async_resume().await {
             GeneratorState::Yielded(block) => {
                 let block_hash = block
```
Comment (thomaseizinger) [inline cnd/src/btsieve/ethereum.rs:147]: This variable is no longer needed now I guess, you can just inline that :)

File: cnd/src/btsieve/ethereum/cache.rs [hunk 3]
```diff
@@ -51,7 +51,7 @@ where
     async fn latest_block(&self) -> anyhow::Result<Self::Block> {
         let block = self.connector.latest_block().await?;
 
-        let block_hash = block.hash.expect("no blocks without hash");
+        let block_hash = block.hash;
```
Comment (thomaseizinger) [inline cnd/src/btsieve/ethereum/cache.rs:54]: Same here, no need for this variable, it can be inlined.

File: cnd/src/btsieve/ethereum.rs (hunk 1 above)
Comment (dhaileytaha) [reply cnd/src/btsieve/ethereum.rs]: done

**Comment by dhaileytaha** [comment]
closed this one and routed to https://github.com/comit-network/comit-rs/pull/2580

**Pull request closed without merging**